libindy/tests/utils/anoncreds.rs

```diff
@@ -454,130 +454,149 @@ pub fn tails_writer_config() -> String {
+                                                                                                         COMMON_MASTER_SECRET).unwrap();
+    //13. Issuer1 Creates XYZ Credential
+    let (issuer1_xyz_cred, _, _) = issuer_create_credential(wallet_handle,
+                                                            &issuer1_xyz_credential_offer,
+                                                            &issuer1_xyz_credential_req,
+                                                            &xyz_credential_values_json(),
+                                                            None,
+                                                            None).unwrap();
+
+    //14. Prover stores Credential
+    prover_store_credential(wallet_handle,
+                            CREDENTIAL2_ID,
+                            &issuer1_xyz_credential_req_metadata,
+                            &issuer1_xyz_cred,
+                            &issuer1_xyz_credential_def_json,
+                            None).unwrap();
+
+    // Issuer2 issues GVT Credential
+    //15. Prover Creates Credential Request
+    let (issuer2_gvt_credential_req, issuer2_gvt_credential_req_metadata) = prover_create_credential_req(wallet_handle,
+                                                                                                         DID_MY1,
+                                                                                                         &issuer2_gvt_credential_offer,
+                                                                                                         &issuer2_gvt_credential_def_json,
+                                                                                                         COMMON_MASTER_SECRET).unwrap();
+
+    //16. Issuer2 Creates XYZ Credential
+    let (issuer2_gvt_cred, _, _) = issuer_create_credential(wallet_handle,
+                                                            &issuer2_gvt_credential_offer,
+                                                            &issuer2_gvt_credential_req,
+                                                            &gvt2_credential_values_json(),
+                                                            None,
+                                                            None).unwrap();
+
+    //17. Prover Stores Credential
+    prover_store_credential(wallet_handle,
+                            CREDENTIAL3_ID,
+                            &issuer2_gvt_credential_req_metadata,
+                            &issuer2_gvt_cred,
+                            &issuer2_gvt_credential_def_json,
+                            None).unwrap();
+
+
+    wallet::close_wallet(wallet_handle).unwrap();
+
+    (issuer1_gvt_credential_def_json,
+     issuer1_gvt_credential_offer,
+     issuer1_gvt_credential_req,
+     issuer1_gvt_cred,
+     config)
+}
+
+pub fn init_common_wallet_once() -> (&'static str, &'static str, &'static str, &'static str) {
```

Comment:
@AxelNennker I don't see usage of this function in the PR. Is it temporary state or you are going to delete it?

Reply:
I think that is a relict of the former code where all tests use _one_ wallet and that is reused. So there is this 'once' stuff that ensures that the wallet is only initialized once.

I kept it because there are some or one test (parallel_auth_encrypt) that tests encryption in parallel and I thought that maybe that test  uses one common wallet. But it turned out that that test also uses separate wallets for the each thread it starts. But I thought maybe we should have a test where several threads work on one wallet in parallel and so I kept this method.

Reply:
@AxelNennker There is critical drawback introduced by current PR:
Wallet preparation for Anoncreds is very long operation and prepare wallet every time from scratch (even in parallel) will increase time of testing significantly.

I'm afraid that benefit from parallel run of tests will be totally eliminated by anoncreds wallet preparations.

Reply:
Problem is that CredentialDefinitions etc are generated without the possibility to NOT store them in a wallet. So they are generated again an again. 
I guess there is nothing we can do - speed things up?
Not sure whether creating one wallet exporting it and importing it into the separate wallets fast enough. Which would leave that one wallet in the indy_home. Maybe data generated during tests should be in the target directory instead of $HOME/.indy_client/ 
I'll look into excluding anoncreds from this PR or ...
I'm open for suggestions. More faster CI/CD servers? Running all the tests locally was never fast...

Reply:
I like the idea to try to export smth and re-use it in tests

Reply:
I'll try something. Although the first merge conflicts appeared. Maybe excluding the anoncread changes from this PR gets us forward and fix the anoncreds stuff later. Breaking this up in smaller chunks might be a good idea anyway.

Reply:
@AxelNennker it's up to you. To accept this PR I see 2 options right now:
1) implement importing wallet for anoncreds tests
2) exclude changes around shared wallet for anoncreds from this PR